src/sql/parts/disasterRecovery/restore/common/restoreServiceImpl.ts

```diff
@@ -153,9 +160,29 @@ export class RestoreDialogController implements IRestoreDialogController {
 			restoreOption.taskExecutionMode = TaskExecutionMode.executeAndScript;
 		}
 
-		this._restoreService.restore(this._ownerUri, restoreOption);
-		let restoreDialog = this._restoreDialogs[this._currentProvider];
-		restoreDialog.close();
+		this._restoreService.restore(this._ownerUri, restoreOption).then(() => {
+			const self = this;
+			let connectionProfile = self._connectionService.getConnectionProfile(self._ownerUri);
+			let activeNode = self._objectExplorerService.getObjectExplorerNode(connectionProfile);
+			this._taskService.onTaskComplete(response => {
```

Comment:
It looks like this code will add this handler every time a restore action is run, but the handler doesn't do any sort of check to verify that the completed task is the restore task. That means that if you restore a database on some connection, then restore a database on a different connection, then after the second restore this code will execute again and you'll refresh both the first and second connections.

That's probably not a huge issue, but the intention is for it only to run for the task that was just kicked off, right? In that case I think some information about the task is returned by the call to `restoreService.restore` above that you might be able to use